tile.go

```diff
@@ -198,3 +212,48 @@ func addExtensions(b *cryptobyte.Builder, leafIndex int64) {
 		b.AddBytes(ext)
 	})
 }
+
+// A TrimmedEntry is a subset of the information in a [LogEntry], including
+// names parsed from the certificate or pre-certificate.
+type TrimmedEntry struct {
+	// Timestamp is the UNIX timestamp in milliseconds of when the entry was
+	// added to the log.
+	Timestamp int64
+
+	// Subject is a DER encoded RDNSequence.
+	//
+	// It is omitted if it includes only a CommonName that matches one of the
+	// DNS or IP entries. That is the case for all Domain Validated WebPKI
+	// certificates.
+	Subject []byte `json:",omitempty"`
+
+	// DNS and IP are the Subject Alternative Names of the certificate.
+	DNS []string `json:",omitempty"`
+	IP  []string `json:",omitempty"`
+}
+
+func (e *LogEntry) TrimmedEntry() (*TrimmedEntry, error) {
+	t := &TrimmedEntry{
+		Timestamp: e.Timestamp,
+	}
+	certBytes := e.Certificate
+	if e.IsPrecert {
+		certBytes = e.PreCertificate
+	}
+	cert, err := x509.ParseCertificate(certBytes)
+	if cert == nil { // x509.ParseCertificate can return non-fatal errors
+		return nil, fmt.Errorf("failed to parse certificate: %w", err)
+	}
+	for _, name := range cert.Subject.Names {
+		if !name.Type.Equal(pkix.OIDCommonName) {
+			t.Subject = cert.RawSubject
+			break
+		}
+	}
```

Comment:
This doesn't quite match the spec, since it doesn't check that the CN matches one of the SANs.

I'm curious how much omitting the Subject actually helps given that the files are compressed. If it's negligible, I would propose just ditching the Subject omission to make the API easier to explain and implement.

Reply:
Of course - I forgot that it would be base64-encoded so you're surely right it would not compress well.

This realization makes me wonder how well the typical consumer of this API will be able to handle a standalone DER-encoded `RDNSequence`. That might be harder with some libraries than parsing an entire certificate, which would deter use of the API. Maybe instead of the DER, it should be an object with the most common subject attributes as strings?

Reply:
Oh! This reminded me that I had planned to make this field in the Go API a custom type with an UnmarshalJSON method and a method that returns a pkix.Name. That solves the usability part if they use the library, it doesn't if they fetch directly, so maybe your suggestion is still better?

Reply:
Why not just always exclude the Subject? In WebPKI, the CN will always be in a SAN, and the rest of the subject is only interesting for OV/EV, which won't have full context without the rest of the certificate anyways.

Reply:
It was a (fairly unsubstantiated) guess that some of the names-only clients do care about associating domain names to organization names, when that information is available. I am not sure how to back that up or disprove it.

Reply:
Switched TrimmedEntry.Subject to an object with the common fields. Including CommonName increases the size by 15% which is probably worth it to avoid explaining CN doesn't matter anymore. It's still less than 4.5% of a full data tile.